hw/top_earlgrey/data/top_earlgrey.hjson

```diff
@@ -631,6 +626,13 @@
       clock_group: "infra",
       reset_connections: {rst_ni: "sys"},
       base_addrs: {rom: "0x00008000", regs: "0x411e0000"}
+      memory: {
```

Comment:
nice, so once i do this to flash also, we can basically remove `memory` completely from top_earlgrey.hjson right? (i'm not sure how happy english_breakfast will be at the moment).

Reply:
In theory yes. There are two things we still need to sort out to get there:
1) as you mentioned, `top_englishbreakfast` still makes use of the "standard" ROM, mainly because it does not have a KMAC instance to implement the actual ROM checking. If the ROM check bypass option is still present, we could technically use that to instantiate a dummy `rom_ctrl` as we've had it in `top_earlgrey` at the beginning, when the KMAC app interface was not yet present.
2) I believe there is some virtual address aliasing going on in the flash controller:
https://github.com/lowRISC/opentitan/blob/ad886b70f91cac3316f15ead30dc9b2f4f5bbf29/util/topgen/templates/toplevel_memory.ld.tpl#L31-L33
We need to figure out how this can be handled with this mechanism.

Reply:
The ROM check bypass option isn't present any more (I removed it happily once everything worked!), but we could put it back in.

Reply:
I think this bypass option could come in handy if we're not planning on adding KMAC to the reduced FPGA config for SCA. This would allow us to better align the two top configuration Hjsons. If it's not difficult to add back in, should we do so?

@tjaychen WDYT?